tiatoolbox/dataloader/wsireader.py

```diff
@@ -82,39 +91,524 @@ def slide_info(self):
+            ...     units="mpp",
+            ... )
 
         """
         raise NotImplementedError
 
-    def slide_thumbnail(self):
-        """Read whole slide image thumbnail at 1.25x.
+    def read_bounds(self, start_w, start_h, end_w, end_h, resolution=0, units="level"):
+        """Read a region of the whole slide image within given bounds.
+
+        Bounds are in terms of the baseline image (level 0  /
+        maximum resolution).
+
+        Reads can be performed at different resolutions by supplying a
+        pair of arguments for the resolution and the
+        units of resolution.
+
+        The output image size may be different
+        to the width and height of the bounds as the resolution will
+        affect this. To read a region with a fixed output image size see
+        :func:`read_rect`.
 
         Args:
-            self (WSIReader):
+            start_w (int): starting point in x-direction (along width).
+            start_h (int): starting point in y-direction (along height).
+            end_w (int): end point in x-direction (along width).
+            end_h (int): end point in y-direction (along height).
+            resolution (int or float or tuple of float): resolution at
+                which to read the image, default = 0. Either a single
+                number or a sequence of two numbers for x and y are
+                valid. This value is in terms of the corresponding
+                units. For example: resolution=0.5 and units="mpp" will
+                read the slide at 0.5 microns per-pixel, and
+                resolution=3, units="level" will read at level at
+                pyramid level / resolution layer 3.
+            units (str): the units of resolution, default = "level".
+                Supported units are: microns per pixel (mpp), objective
+                power (power), pyramid / resolution level (level),
+                pixels per baseline pixel (baseline).
 
         Returns:
-            ndarray : image array
+            ndarray : array of size MxNx3
+            M=end_h-start_h, N=end_w-start_w
 
+        Examples:
+            >>> from tiatoolbox.dataloader import wsireader
+            >>> from matplotlib import pyplot as plt
+            >>> wsi_obj = wsireader.WSIReader("/path/to/a/wsi")
+            >>> # Read a region at level 0 (baseline / full resolution)
+            >>> bounds = [1000, 2000, 2000, 3000]
+            >>> img = wsi_obj.read_bounds(*bounds)
+            >>> plt.imshow(img)
+            >>> # This could also be written more verbosely as follows
+            >>> img = wsi_obj.read_bounds(
+            ...     start_w=bounds[0],
+            ...     start_h=bounds[1],
+            ...     end_w=bounds[2],
+            ...     end_h=bounds[3],
+            ...     resolution=0,
+            ...     units="level",
+            ... )
+            >>> plt.imshow(img)
+
+        Note: The field of view remains the same as resolution is varied.
```

Comment:
Can we say, "Note: The field of view remains the same as resolution is varied when using wsi.read_bounds()." just to make it self-complete. On docs it requires a bit of scrolling.

Reply:
Now done in commit https://github.com/TIA-Lab/tiatoolbox/pull/40/commits/427256fa619cf84d3d72759ec8ba77bdfc695f6f